src/client/datascience/dataScienceSurveyBanner.ts

```diff
@@ -140,7 +140,8 @@ export class DataScienceSurveyBanner implements IJupyterExtensionBanner {
         if (!show) {
             return;
         }
-
+        // Disable for the current session.
+        this.disabledInCurrentSession = true;
```

Comment:
Is this the right location? It's still after an await statement, and also after the disabledInCurrentSession check. So can't two calls to show banner both get released?

Reply:
Ahh, the shouldShow didn't need to be async. Yeah, that looks good now.